app/models/concerns/order_shipping_method.rb

```diff
@@ -18,4 +18,17 @@ def shipping_method
     return if shipments.empty?
     shipments.first.shipping_method
   end
+
+  # Finds the shipment shipping_rate for the given shipping_method_id and selects that shipping_rate
+  #
+  # @return [Shipment]
+  def select_shipping_method(shipping_method_id)
+    return if shipments.empty?
+    shipment = shipments.first
+
+    shipping_rate = shipment.shipping_rates.find{ |shipping_rate| shipping_rate.shipping_method_id == shipping_method_id }
+    return if !shipping_rate
+
+    shipment.selected_shipping_rate_id=(shipping_rate.id)
```

Comment:
Is this the way Spree does it?

Reply:
no, in Spree's multi-step checkout process, the shipping_rates are rendered on the UI and the user selection is done on shipping_rates. We are not doing the same, we will let the user select a shipping_method and we will inject the selection on the shipping_rate ourselves. We could do the same even with single step checkout with some ajax call but we would have to change the rendering of the checkout page. I dont think this is necessary right now.